modules/services/preferences/src/main/java/au/com/shiftyjelly/pocketcasts/preferences/SettingsImpl.kt

```diff
@@ -339,6 +339,26 @@ class SettingsImpl @Inject constructor(
         return timeSinceDismiss >= 7.days
     }
 
+    override fun setDismissedSuggestedFolderPaywallTime() {
+        sharedPreferences.edit { putLong(Settings.LAST_DISMISS_SUGGESTED_FOLDER_PAYWALL_TIME, System.currentTimeMillis()) }
+    }
+
+    override fun isEligibleToShowSuggestedFolderPaywall(): Boolean {
+        val lastDismissedTime = sharedPreferences.getLong(Settings.LAST_DISMISS_SUGGESTED_FOLDER_PAYWALL_TIME, 0L)
+        val currentDismissedCount = sharedPreferences.getInt(Settings.DISMISS_SUGGESTED_FOLDER_PAYWALL_COUNT, 0)
+
+        if (lastDismissedTime == 0L) return true
+
+        val timeSinceDismiss = (System.currentTimeMillis() - lastDismissedTime).milliseconds
+
+        return timeSinceDismiss >= 7.days && currentDismissedCount < 2
```

Comment:
I could be wrong but would business logic like this be better outside of the settings class? 🤔